src/ui/widgets/calendar.rs

```diff
@@ -0,0 +1,111 @@
+use super::*;
+
+const DAY_SIZE: (u16, u16) = (4, 3);
+
+pub struct CalendarWidget {
+    year: i32,
+    month: u32,
+}
+
+impl Default for CalendarWidget {
+    fn default() -> Self {
+        let now = Local::now();
+        CalendarWidget {
+            year: now.year(),
+            month: now.month(),
+        }
+    }
+}
+
+impl Widget for CalendarWidget {
+    fn render(self, mut area: Rect, buf: &mut Buffer) {
+        let bounds = self.widget_bounds();
+        if area.width < bounds.width || area.height < bounds.height {
+            return; // TODO: figure out what to do when area is to small
+        }
+        area = self.align_area(area, bounds);
+
+        let cols = Layout::default()
+            .direction(Direction::Horizontal)
+            .constraints([Constraint::Length(DAY_SIZE.0); 7])
+            .split(area);
+
+        let mut constraints = [Constraint::Length(DAY_SIZE.1); 7];
```

Comment:
this should be struct
struct DayConstraints {
    header: Constraint,
    body: [Constraint; 7],
}

more readable in my opinion, what do you think?

Reply:
sure ok.